tests/nvmeof/test_ceph_nvmeof_high_availability.py

```diff
@@ -0,0 +1,313 @@
+                "command": "remove",
+                "service": "nvmeof",
+                "args": {"service_name": f"nvmeof.{config['rbd_pool']}"},
+            },
+        }
+        test_orch.run(ceph_cluster, **cfg)
+
+    # Delete the pool
+    if "pool" in config["cleanup"]:
+        rbd_obj.clean_up(pools=[config["rbd_pool"]])
+
+
+def run(ceph_cluster: Ceph, **kwargs) -> int:
+    """Return the status of the Ceph NVMEof HA test execution.
+
+    - Configure SPDK and install with control interface.
+    - Configures Initiators and Run FIO on NVMe targets.
+
+    Args:
+        ceph_cluster: Ceph cluster object
+        kwargs: Key/value pairs of configuration information to be used in the test.
+
+    Returns:
+        int - 0 when the execution is successful else 1 (for failure).
+
+    Example:
+
+        # Execute the nvmeof GW test
+            - test:
+                name: Ceph NVMeoF deployment
+                desc: Configure NVMEoF gateways and initiators
+                config:
+                    gw_node: node6
+                    rbd_pool: rbd
+                    do_not_create_image: true
+                    rep-pool-only: true
+                    cleanup-only: true                          # only for cleanup
+                    rep_pool_config:
+                      pool: rbd
+                    install: true                               # Run SPDK with all pre-requisites
+                    subsystems:                                 # Configure subsystems with all sub-entities
+                      - nqn: nqn.2016-06.io.spdk:cnode3
+                        serial: 3
+                        bdevs:
+                          count: 1
+                          size: 100G
+                        listener_port: 5002
+                        allow_host: "*"
+                    failover: systemctl-node                    # fail-tool: systemctl, nodes-to-be-failed: node names
+    """
+    LOG.info("Starting Ceph Ceph NVMEoF deployment.")
+    config = kwargs["config"]
+    rbd_pool = config["rbd_pool"]
+    rbd_obj = initial_rbd_config(**kwargs)["rbd_reppool"]
+
+    overrides = kwargs.get("test_data", {}).get("custom-config")
+    for key, value in dict(item.split("=") for item in overrides).items():
+        if key == "nvmeof_cli_image":
+            NVMeGWCLI.NVMEOF_CLI_IMAGE = value
+            break
+
+    gw_nodes = get_nodes_by_ids(ceph_cluster, config["gw_nodes"])
+    gw_port = config.get("gw_port", 5500)
+    nvmegwcli = NVMeGWCLI(gw_nodes[0], gw_port)
+
+    try:
+        if config.get("install"):
+            cfg = {
+                "no_cluster_state": False,
+                "config": {
+                    "command": "apply",
+                    "service": "nvmeof",
+                    "args": {"placement": {"nodes": [i.hostname for i in gw_nodes]}},
+                    "pos_args": [rbd_pool],
+                },
+            }
+            test_nvmeof.run(ceph_cluster, **cfg)
```

Comment:
So as of now we don't have the option to specify the cpu_mask parameter is it?

Reply:
In downstream, by default cpu_mask is set to `4`, so far any other test case requirement then it has to be developed as per test case.